internal/json-anonymizer/json-anonymizer.go

```diff
@@ -147,18 +147,18 @@ func (a *JsonAnonymizer) buildJqQuery() (query string, transformerFunctions []st
 	queryParts := []string{}
 	functionNames := []string{}
 
+	if a.defaultTransformers != nil {
```

Comment:
moved this out of the for loop. Guessing this reduced the amount of query parts added? Did this solve a bug or anything?

Reply:
Apply defaults was being run last and that causes any mappings in the transformer mappings that are using an expression to be overwritten (paths are okay because we know to skip them). If apply default is run first then expressions applied second it allows users to use more specific generators with expressions. This came up because I was writing the docs

ex
before change:
1. apply expression transformer generate first name to all name fields
2. apply default transformers for string character scramble
this case user ends up with only character scramble fields

after
1. apply default transformers for string character scramble
2. apply expression transformer generate first name to all name fields
users ends up with name fields that look like real names, with the caveat that they have been transformed twice

Reply:
Won't that cause any configs that are expecting to run Transformers (instead of Generators) on input data to transform generated data instead of the original data?

Reply:
yes that's correct but only when using JQ expressions. The real fix is figuring out all the paths hit by expressions

Reply:
Sounds good

Reply:
made a ticket
https://linear.app/nucleus-cloud/issue/NEOS-1521/json-anonymizer-jq-expressions-find-all-paths-and-add-them-to-skip